library/src/ui/index.js

```diff
@@ -85,4 +88,162 @@ window.CoNDeT.ui = {
+    };
+    constructor.prototype.onUpdate = function (common, props) {
+      this.ref.innerHTML = props.text;
+    };
+    constructor.prototype.onDestroy = function (common) {
+      this.ref.remove();
+    };
+
+    return constructor;
+  })(),
+
+  Head: (function () {
+    function constructor() {}
+
+    constructor.prototype.appendHeaders = function (headerRef, content) {
+      for (let i = 0; i < content.conditions.length; i++) {
+        let headerCol = document.createElement("th");
+        headerCol.classList.add("condition");
+        headerCol.innerHTML = content.conditions[i];
+        headerRef.appendChild(headerCol);
+      }
+
+      for (let i = 0; i < content.decisions.length; i++) {
+        let headerCol = document.createElement("th");
+        headerCol.classList.add("decision");
+        headerCol.innerHTML = content.decisions[i];
+        headerRef.appendChild(headerCol);
+      }
+    };
+
+    constructor.prototype.onInit = function (common, props) {
+      this.ref = document.createElement("thead");
+      this.headerRow = document.createElement("tr");
+      this.ref.appendChild(this.headerRow);
+
+      this.appendHeaders(this.headerRow, {
+        conditions: props.conditions,
+        decisions: props.decisions,
+      });
+    };
+    constructor.prototype.onUpdate = function (common, props) {
+      for (let i = 0; i < this.ref.children.length; i++) {
+        this.ref.removeChild(this.ref.children[i]);
+      }
+      this.appendHeaders(this.headerRow, {
+        conditions: props.conditions,
+        decisions: props.decisions,
+      });
+    };
+    constructor.prototype.onDestroy = function (common) {
+      this.ref.remove();
+    };
+
+    return constructor;
+  })(),
+
+  Body: (function () {
+    function constructor() {}
+
+    constructor.prototype.onInit = function (common, props) {
+      this.ref = document.createElement("tbody");
+      this.children = [];
+
+      for (let i = 0; i < props.content.length; i++) {
+        this.createChild(Row, { constent: props.content[i] });
+      }
+    };
+    constructor.prototype.onUpdate = function (common, props) {
+      for (let i = 0; i < this.children.length; i++) {
+        this.children[i].onDestroy(common);
+      }
+      for (let i = 0; i < props.content.length; i++) {
+        this.createChild(Row, { constent: props.content[i] });
+      }
+    };
+    constructor.prototype.onDestroy = function (common) {
+      this.ref.remove();
+    };
+
+    return constructor;
+  })(),
+
+  Row: (function () {
+    function constructor() {}
+
+    constructor.prototype.onInit = function (common, props) {
+      this.ref = document.createElement("tr");
+
+      for (let i = 0; i < props.content.length; i++) {
+        let cell = document.createElement("td");
+        cell.innerHTML = props.content[i];
+        this.ref.appendChild(cell);
+      }
+    };
+
+    constructor.prototype.onUpdate = function (common, props) {};
```

Comment:
Myślałem, czy row nie może się zmienić, ale usuwasz wyżej wszystkie wiersze i wstawiasz nowe, więc git. Ale to nie jest optymalne i jak będzie lagować, to to jest pierwsza rzecz, która idzie pod refactor